tools/wptrunner/wptrunner/browsers/sauce.py

```diff
@@ -156,10 +156,19 @@ def __enter__(self, options):
             "web-platform.test",
             "*.web-platform.test"
         ])
-        while not os.path.exists('./sauce_is_ready') and not self.sc_process.poll():
-            time.sleep(5)
 
-        if self.sc_process.returncode is not None and self.sc_process.returncode > 0:
+        tot_wait = 0
+        while not os.path.exists('./sauce_is_ready') and self.sc_process.poll() is None:
+            time.sleep(5)
```

Comment:
Put this in a variable and use it n your counter.

Reply:
Does this look better now?